### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by ragingpastry**
Add a dockerfile for docker deployments.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+RUN pip install virtualenv
+RUN virtualenv /build
+RUN apk add --update --no-cache \
+    python \
+    python-dev \
+    build-base \
+    linux-headers \
+    pcre-dev \
+    py-pip \
+    curl \
```
Comment (dingus9) [inline Dockerfile]: do we use curl?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    openssl \
+    git
```
Comment (dingus9) [inline Dockerfile:9]: that's odd to have git inside a container

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+FROM python:3.7-alpine as build
+WORKDIR /build
+RUN pip install virtualenv
+RUN virtualenv /build
+RUN apk add --update --no-cache \
+    python \
+    python-dev \
+    build-base \
+    linux-headers \
+    pcre-dev \
```
Comment (dingus9) [inline Dockerfile:8]: what's this for? when we have python?

File: uwsgi.yml [hunk 4]
```diff
@@ -5,16 +5,17 @@ uwsgi:
...
+  #chroot: /tmp/uwsgi
+  gid: uwsgi
+  uid: uwsgi
 
   ## stats ##
   # stats: /tmp/statsock
 
   ## debug stuff ##
-  # honour-stdin: true
+  honour-stdin: true
```
Comment (dingus9) [inline uwsgi.yml]: we'll probably need to change this to false for prod

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+COPY requirements.txt /build
+RUN /build/bin/pip install -r requirements.txt uwsgi
+
+FROM python:3.7-alpine as application
+RUN adduser -D -g '' uwsgi
+RUN apk add --no-cache git
+WORKDIR /app
+COPY --from=build /build /build
+COPY --from=build /usr/lib /usr/lib
+COPY . /app/
```
Comment (dingus9) [inline Dockerfile]: we should be able to install gman with pip instead of copying locally actually

**Review by dingus9 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ragingpastry) [reply Dockerfile]: Nope. can get it out

File: Dockerfile (hunk 2 above)
Comment (ragingpastry) [reply Dockerfile:9]: We need git to install piperci-python

File: Dockerfile (hunk 5 above)
Comment (ragingpastry) [reply Dockerfile]: Sure we can make that change

**Review by dingus9 — APPROVED** [review]
Lets try it

**Pull request merged**